pkg/apis/core/validation/controllerdeployment.go

```diff
@@ -58,9 +59,48 @@ func ValidateControllerDeploymentUpdate(new, _ *core.ControllerDeployment) field
 func validateHelmControllerDeployment(helmControllerDeployment *core.HelmControllerDeployment, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 
-	// rawChart is required as long as there is no other source (OCI sources will be added later on)
-	if len(helmControllerDeployment.RawChart) == 0 {
-		allErrs = append(allErrs, field.Required(fldPath.Child("rawChart"), ""))
+	if (len(helmControllerDeployment.RawChart) == 0) == (helmControllerDeployment.OCIRepository == nil) {
+		allErrs = append(allErrs, field.Required(fldPath, "must provide either rawChart or ociRepository"))
+	}
+
+	allErrs = append(allErrs, validateOCIRepository(helmControllerDeployment.OCIRepository, fldPath.Child("ociRepository"))...)
+
+	return allErrs
+}
+
+func validateOCIRepository(oci *core.OCIRepository, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if oci == nil {
+		return allErrs
+	}
+
+	if oci.Ref == "" && oci.Repository == "" {
+		allErrs = append(allErrs, field.Required(fldPath, "must provide either ref or repository"))
```

Comment:
```suggestion
		allErrs = append(allErrs, field.Required(fldPath, "must provide either ref or repository"))
		return allErrs
```

Otherwise, `oci.Repository` are showed in addition which can be confusing.